### PR TITLE
Do not fail container creation if the contaienr exits before applyting

### LINKED ISSUE #20017: Add version to e2e.test binary artifact.
We will want to ship the e2e binary w/ the ability to independently verify its version, esp pending #19624 .
cc @timothysc @liggitt @ixdy @pmorie @kubernetes/goog-testing .  

I'll craft a PR for this after looking at how version is handled in kubectl, im sure it will be quite simple.

### LINKED ISSUE #20117: Kubelet fails to distinguish failure in running container from a quickly exiting container
My (partly unfounded) theory is that we should refactor `ApplyOOMScoreAdjContainer` out of `runContainerInPod` because if the container exited real quick it won't have an entry in `/proc/pid/cgroup` (https://github.com/kubernetes/kubernetes/blob/d3b869ae14db522e5fca91a295b8bc870713013f/pkg/kubelet/dockertools/manager.go#L1537), and we confuse this for an error in running the container.

This is based off the following debugging session:
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-parallel/14712/ 
`pod-995d097b-bfc2-11e5-9ce3-42010af01555_e2e-tests-emptydir-fbgqp` has `restart=Never` and it seems to exit, and we don't report it as failed/succeeded. 

1. Started container:
I0120 22:10:22.967700    3461 server.go:569] Event(api.ObjectReference{Kind:"Pod", 
Namespace:"e2e-tests-emptydir-fbgqp", Name:"__pod-995d097b-bfc2-11e5-9ce3-42010af01555__", 
UID:"995d410d-bfc2-11e5-9d46-42010af00002", APIVersion:"v1", ResourceVersion:"8929", 
FieldPath:"spec.containers{test-container}"}): type: 'Normal' reason: 'Started' Started container with 
docker id fab06c2e3ee6

2. Failed to set oom score:
__E0120 22:10:23.007292__    3461 manager.go:1890] Error running pod "pod-995d097b-bfc2-11e5-9ce3-42010af01555_e2e-tests-emptydir-fbgqp(995d410d-bfc2-11e5-9d46-42010af00002)" __container "test-container": open /proc/17193/cgroup: no such file or directory__

3. FailedSync event:
I0120 22:10:23.070581    3461 server.go:569] Event(api.ObjectReference{Kind:"Pod", 
Namespace:"e2e-tests-emptydir-fbgqp", Name:"pod-995d097b-bfc2-11e5-9ce3-42010af01555", 
UID:"995d410d-bfc2-11e5-9d46-42010af00002", APIVersion:"v1", ResourceVersion:"8929", 
FieldPath:""}): type: 'Warning' reason: 'FailedSync' Error syncing pod, skipping: not all containers have 
started: 0 != 1

4. However if you track the next 2 container inspect results: 

  * I0120 22:10:23.053457    3461 manager.go:340] Container inspect result: 
{ID:__fab06c2e3ee6eaff383bbb3d9f77ded5d692e20e7c77d849bdfdfdc6fe475baf__ Created:2016-01-20 
22:10:22.821645807 +0000 UTC Path:/mt Args:[--fs_type=/test-volume --new_file_0644=/test-
volume/test-file --file_perm=/test-volume/test-file] Config:0xc208408fc0 State:{Running:true 
Paused:false Restarting:false OOMKilled:false __Pid:17193 ExitCode:0 Error: StartedAt:2016-01-20 
22:10:22.967310973 +0000 UTC FinishedAt:0001-01-01 00:00:00 +0000 UTC}__ 
Image

  * I0120 22:10:23.328738    3461 manager.go:340] Container inspect result: 
{ID:__fab06c2e3ee6eaff383bbb3d9f77ded5d692e20e7c77d849bdfdfdc6fe475baf__ Created:2016-01-20
 22:10:22.821645807 +0000 UTC Path:/mt Args:[--fs_type=/test-volume --new_file_0644=/test-
volume/test-file --file_perm=/test-volume/test-file] Config:0xc208748380 State:{Running:false 
Paused:false Restarting:false OOMKilled:false Pid:0 __ExitCode:0 Error: StartedAt:2016-01-20
22:10:22.967310973 +0000 UTC FinishedAt:2016-01-20 22:10:23.06969655 +0000 
UTC}__ ResolvConfPath:

5. And we sent the Running/Ready update but no status update after that:
I0120 22:10:23.080449    3461 manager.go:371] Status for pod "pod-995d097b-bfc2-11e5-9ce3-42010af01555_e2e-tests-emptydir-fbgqp(995d410d-bfc2-11e5-9d46-42010af00002)" updated successfully: {status:{Phase:Running Conditions:[{Type:Ready Status:True LastProbeTime:{Time:{sec:0 nsec:0 loc:0x2bba980}} LastTransitionTime:{Time:{sec:63588924623 nsec:0 loc:0x2bba980}} Reason: Message:}] Message: Reason: HostIP:10.240.0.5 PodIP:10.248.84.65 

@kubernetes/goog-node thoughts?

### PR DESCRIPTION
**Pull request by vishh**
oom score adjust.

Fixes #20117 

cc @dchen1107

### CODE REVIEW

**Comment by dchen1107** [comment]
I guess you are referring to a wrong issue above, but I like what you proposed here. cc/ @bprashanth since he reported an issue related to this.

**Comment by bprashanth** [comment]
I think reading /proc is racy and should be non-critical, and GetFullContainerName will invoke readfile on a non-existent file and return an error: https://github.com/kubernetes/kubernetes/blob/133440188822cbec7f9784498e2bc372de2f33cc/pkg/util/procfs/procfs.go#L49

Fwiw, when I thought of a fix I wanted to do 2 things:
1. Make reading /proc always non-critical because of the racyness
2. Move oom score adjustements into something periodic that we keep trying on every sync

**Comment by vishh** [comment]
Your suggestions make total sense. Fixing `1` is straightforward. However
the current state of kubelet makes adding any periodic housekeeping to pods
harder. So `2` requires a small surgery to kubelet, which I'm going to
avoid for now, because this logic will be deprecated once we officially
support docker v1.10.

**Comment by dchen1107** [comment]
LGTM

File: pkg/util/oom/oom_linux.go [hunk 1]
```diff
@@ -60,20 +73,22 @@ func applyOOMScoreAdj(pid int, oomScoreAdj int) error {
...
 	for i := 0; i < maxTries; i++ {
-		_, readErr := ioutil.ReadFile(oomScoreAdjPath)
-		if readErr != nil {
-			err = fmt.Errorf("failed to read oom_score_adj: %v", readErr)
-		} else if writeErr := ioutil.WriteFile(oomScoreAdjPath, []byte(strconv.Itoa(oomScoreAdj)), 0700); writeErr != nil {
-			err = fmt.Errorf("failed to set oom_score_adj to %d: %v", oomScoreAdj, writeErr)
-		} else {
-			return nil
+		if err = ioutil.WriteFile(oomScoreAdjPath, []byte(value), 0700); err != nil {
+			exists, perr := processExists(pid)
```
Comment (bprashanth) [inline pkg/util/oom/oom_linux.go]: does the caller of this function (applyoomscoreadjcontainer) require similar treatement or is it absolved? Noticed it translates cgroups to pids by peeking in sysfs (I think). Or is that not the right callstack even.

Comment (vishh) [inline pkg/util/oom/oom_linux.go]: Good point. Fixed that.

**Comment by dchen1107** [comment]
LGTM

**Comment by dchen1107** [comment]
@vishh please ping me once you have done rebasing.

**Comment by dchen1107** [comment]
LGTM again

**Comment by vishh** [comment]
@k8s-bot: test again #20646

**Comment by vishh** [comment]
@k8s-bot: test this #20646

**Comment by vishh** [comment]
@k8s-bot: test this #20646

**Comment by ArtfulCoder** [comment]
@k8s-bot test this

**Comment by vishh** [comment]
@ArtfulCoder: I'm not sure why you were mentioned in this PR.

**Comment by vishh** [comment]
@k8s-bot: test this issue #20646

**Comment by vishh** [comment]
@dchen1107: I'm re-adding the LGTM label since the bot removed it for no apparent reason (the patch hasn't changed since your last LGTM)

**Pull request merged**